examples/gitserver/README.md

```diff
@@ -1,16 +1,220 @@
+   ```sh
+   $ oc edit bc ruby-hello-world
+   ```
+   For an example, see Step 3 here: https://docs.openshift.org/latest/dev_guide/builds.html#basic-authentication
+
+3. Start a new build
+
+   A build can be started explicitly
+
+   ```sh
+   $ oc start-build ruby-hello-world
+   ```
+
+   Or by pushing new commits to your Git Server
+   ```sh
+   $ git commit --amend
+   $ git push origin master
+   ```
+
+### Secure the Git Server
+
+1. If your cluster's router has been setup with a default certificate for edge termination,
+   just modify the route to use edge termination:
+
+   ```sh
+   $ oc edit route gitserver
+   ```
+
+   Add tls -> termination -> edge to the route specification:
+
+   ```yaml
+   apiVersion: v1
+   kind: Route
+   metadata:
+     name: gitserver
+   spec:
+     host: gitserver-myproject.infra.openshift.com
+     tls:
+       termination: edge
+     to:
+       kind: Service
+       name: gitserver
+   ```
+
+2. If using a private certificate authority, configure your git client to use the private ca.crt file:
+
+   ```sh
+   $ git config --global http.https://gitserver-myproject.infra.openshift.com.sslCAInfo /path/to/ca.crt
+   ```
+
+   where the key is http.[git server public URL].sslCAInfo
+
+Authentication
+--------------
+
+By default, the gitserver will authenticate using OpenShift user or service account credentials. For a user,
+the credentials are the user name, and the user's token (from `oc whoami -t`). For a service account, it can
+be one of the 2 tokens created with the service account and stored in the service account secrets. These can 
+be obtained by looking at the secrets that correspond to the service account (`oc get secrets`) and displaying
+the token in one of them: `oc describe secret secret-name`.
+
+
+Authorization
+-------------
+
+Users or service accounts must be able to get pods in the current namespace in order to fetch repositories from
+the Git Server. Specifying ALLOW_ANON_GIT_PULL=yes as an environment variable for the Git Server will allow anyone
+to fetch/clone content from the Git Server. To create/push content, users or service accounts must have the right 
+to create pods in the namespace.
+
+
+
+Automatically Cloning Public Repositories
+-----------------------------------------
+
+The Git Server can automatically clone a set of repositories on startup if it is going to be used for mirroring
+purposes. Repositories to be cloned can be specified using a set of environment variables that match
+`GIT_INITIAL_CLONE_[name]` where [name] must be unique. The value of the variable must be a URL to the remote
+repository to clone and an optional name for the clone.
+
+The following command will add an environment variable to clone the openshift/ruby-hello-world repository and give
+it a name of `helloworld` inside the Git Server.
+
+```sh
+$ oc env dc/gitserver GIT_INITIAL_CLONE_1=https://github.com/openshift/ruby-hello-world.git;helloworld
```

Comment:
since you've got auto-deploy on config change, and an empty dir volume, users should be aware that this is going to destroy all their existing git repos.